controllers/machinehealthcheck_controller.go

```diff
@@ -256,12 +258,63 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for _, t := range unhealthy {
 		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
 
-		if annotations.IsPaused(cluster, t.Machine) {
-			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+		if m.Spec.RemediationTemplate != nil {
+			// If external remediation request already exists,
+			// return early
+			if r.externalRemediationRequestExists(ctx, m, t.Machine.Name) {
+				logger.V(2).Info("External Remediation Request already exists", "machine", t.Machine.Name, "templateGVK", m.Spec.RemediationTemplate.GroupVersionKind())
+				return ctrl.Result{}, nil
+			}
+
+			cloneOwnerRef := &metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       t.Machine.Kind,
+				Name:       t.Machine.Name,
+				UID:        t.Machine.UID,
+			}
+
+			from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+			if err != nil {
+				conditions.MarkFalse(m, clusterv1.ExternalRemediationTemplateAvailable, clusterv1.ExternalRemediationTemplateNotFound, clusterv1.ConditionSeverityError, "External Remediation Template not found")
+				return ctrl.Result{}, errors.Wrapf(err, "error retrieving remediation template %v %q for machine %q in namespace %q within cluster %q", m.Spec.RemediationTemplate.GroupVersionKind(), m.Spec.RemediationTemplate.Name, t.Machine.Name, t.Machine.Namespace, m.Spec.ClusterName)
+			}
+
+			generateTemplateInput := &external.GenerateTemplateInput{
+				Template:    from,
+				TemplateRef: m.Spec.RemediationTemplate,
+				Namespace:   t.Machine.Namespace,
+				ClusterName: t.Machine.ClusterName,
+				OwnerRef:    cloneOwnerRef,
+			}
+			to, err := external.GenerateTemplate(generateTemplateInput)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "failed to create template for remediation request %v %q for machine %q in namespace %q within cluster %q", m.Spec.RemediationTemplate.GroupVersionKind(), m.Spec.RemediationTemplate.Name, t.Machine.Name, t.Machine.Namespace, m.Spec.ClusterName)
```

Comment:
I'd like to see this error visible in a condition.

Reply:
Should we create a new condition for it or can we use `ExternalRemediationTemplateAvailable`?